sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonIUD.java

```diff
@@ -155,7 +158,15 @@ public void update(String path, Expression filterExpression,
     Schema schema = CarbonSchemaReader.readSchema(indexFiles.get(0)).asOriginOrder();
     Field[] fields = schema.getFields();
     String[] projectionColumns = new String[fields.length + 1];
+    List<Integer> dateIndexes = new ArrayList<>();
+    List<Integer> timeStampIndexes = new ArrayList<>();
     for (int i = 0; i < fields.length; i++) {
+      if (fields[i].getDataType() == DataTypes.DATE) {
+        dateIndexes.add(i);
+      }
+      if (fields[i].getDataType() == DataTypes.TIMESTAMP) {
```

Comment:
```suggestion
      else if (fields[i].getDataType() == DataTypes.TIMESTAMP) {
```
and format the line 167 to 166

Reply:
Code removed